diff -ur awn-extras-applets-0.2.6.orig/src/plugger/applet.c awn-extras-applets-0.2.6/src/plugger/applet.c
--- awn-extras-applets-0.2.6.orig/src/plugger/applet.c	2008-03-14 21:39:30.000000000 +0100
+++ awn-extras-applets-0.2.6/src/plugger/applet.c	2008-03-14 21:47:53.000000000 +0100
@@ -36,6 +36,7 @@
 
 #define STACKS_APPLET "/usr/lib/awn/applets/stacks.desktop"
 #define STACKS_APPLET_LOCAL "/usr/local/lib/awn/applets/stacks.desktop"
+#define STACKS_APPLET_SHARE "/usr/share/awn/applets/stacks.desktop"
 
 typedef struct {
   AwnApplet *applet;
@@ -229,10 +230,12 @@
     desktop_path = STACKS_APPLET;
   }else if(gnome_vfs_uri_exists(gnome_vfs_uri_new(STACKS_APPLET_LOCAL))){
     desktop_path = STACKS_APPLET_LOCAL;
+  }else if(gnome_vfs_uri_exists(gnome_vfs_uri_new(STACKS_APPLET_SHARE))){
+    desktop_path = STACKS_APPLET_SHARE;
   }else{
     g_print("!! Stacks Plugger Error: dependency on Stacks Applet not met:\n \
              !! Could not find stacks.desktop file at:\n \
-             !! %s or %s\n", STACKS_APPLET, STACKS_APPLET_LOCAL);
+             !! %s or %s\n", STACKS_APPLET, STACKS_APPLET_LOCAL, STACKS_APPLET_SHARE);
   }
 
   // Create a box that will hold the stacks applets
Seulement dans awn-extras-applets-0.2.6/src/plugger: applet.c~
diff -ur awn-extras-applets-0.2.6.orig/src/trasher/applet.c awn-extras-applets-0.2.6/src/trasher/applet.c
--- awn-extras-applets-0.2.6.orig/src/trasher/applet.c	2008-03-14 21:39:30.000000000 +0100
+++ awn-extras-applets-0.2.6/src/trasher/applet.c	2008-03-14 21:52:22.000000000 +0100
@@ -35,6 +35,7 @@
 
 #define STACKS_APPLET "/usr/lib/awn/applets/stacks.desktop"
 #define STACKS_APPLET_LOCAL "/usr/local/lib/awn/applets/stacks.desktop"
+#define STACKS_APPLET_SHARE "/usr/share/awn/applets/stacks.desktop"
 
 typedef struct {
   AwnApplet *applet;
@@ -120,10 +121,12 @@
     desktop_path = STACKS_APPLET;
   }else if(gnome_vfs_uri_exists(gnome_vfs_uri_new(STACKS_APPLET_LOCAL))){
     desktop_path = STACKS_APPLET_LOCAL;
+  }else if(gnome_vfs_uri_exists(gnome_vfs_uri_new(STACKS_APPLET_SHARE))){
+    desktop_path = STACKS_APPLET_SHARE;
   }else{
     g_print("!! Stacks Trasher Error: dependency on Stacks Applet not met:\n \
              !! Could not find stacks.desktop file at:\n \
-             !! %s or %s\n", STACKS_APPLET, STACKS_APPLET_LOCAL);
+             !! %s or %s\n", STACKS_APPLET, STACKS_APPLET_LOCAL, STACKS_APPLET_SHARE);
   }
 
   // Create a box that will hold the stacks applets
Seulement dans awn-extras-applets-0.2.6/src/trasher: applet.c~
