diff -urpN awn-extras-applets-0.2.6.orig/src/affinity/aff-start.c awn-extras-applets-0.2.6/src/affinity/aff-start.c
--- awn-extras-applets-0.2.6.orig/src/affinity/aff-start.c	2008-02-14 09:49:53.000000000 -0700
+++ awn-extras-applets-0.2.6/src/affinity/aff-start.c	2008-04-20 23:22:01.000000000 -0600
@@ -31,6 +31,7 @@
 #include <glib/gi18n.h>
 #include <libawn/awn-desktop-item.h>
 #include <libawn/awn-vfs.h>
+#include <libgnomevfs/gnome-vfs-utils.h>
 
 #include "aff-start.h"
 
diff -urpN awn-extras-applets-0.2.6.orig/src/affinity/aff-utils.c awn-extras-applets-0.2.6/src/affinity/aff-utils.c
--- awn-extras-applets-0.2.6.orig/src/affinity/aff-utils.c	2008-02-14 09:49:53.000000000 -0700
+++ awn-extras-applets-0.2.6/src/affinity/aff-utils.c	2008-04-21 09:52:48.000000000 -0600
@@ -32,6 +32,7 @@
 #ifdef LIBAWN_USE_GNOME
 #include <libgnomeui/gnome-icon-lookup.h>
 #include <libgnomeui/gnome-thumbnail.h>
+#include <libgnomevfs/gnome-vfs-utils.h>
 #else
 #include "egg-pixbuf-thumbnail.h"
 #ifdef LIBAWN_USE_XFCE
diff -urpN awn-extras-applets-0.2.6.orig/src/awnterm/applet.c awn-extras-applets-0.2.6/src/awnterm/applet.c
--- awn-extras-applets-0.2.6.orig/src/awnterm/applet.c	2008-02-14 09:49:01.000000000 -0700
+++ awn-extras-applets-0.2.6/src/awnterm/applet.c	2008-04-21 09:59:20.000000000 -0600
@@ -21,6 +21,7 @@
 
 #include <libawn/awn-applet.h>
 #include <libawn/awn-applet-simple.h>
+#include <libawn/awn-applet-dialog.h>
 #include <vte/vte.h>
 #include <gtk/gtk.h>
 
diff -urpN awn-extras-applets-0.2.6.orig/src/cairo-menu/backend-gnome.c awn-extras-applets-0.2.6/src/cairo-menu/backend-gnome.c
--- awn-extras-applets-0.2.6.orig/src/cairo-menu/backend-gnome.c	2008-02-14 09:49:53.000000000 -0700
+++ awn-extras-applets-0.2.6/src/cairo-menu/backend-gnome.c	2008-04-20 23:22:01.000000000 -0600
@@ -34,6 +34,7 @@
 #include <libgen.h>
 #include <ctype.h>
 #include <libnotify/notify.h>
+#include <glib/gstdio.h>
 
 #include "menu_list_item.h"
 
diff -urpN awn-extras-applets-0.2.6.orig/src/cairo-menu/render.h awn-extras-applets-0.2.6/src/cairo-menu/render.h
--- awn-extras-applets-0.2.6.orig/src/cairo-menu/render.h	2008-02-14 09:49:01.000000000 -0700
+++ awn-extras-applets-0.2.6/src/cairo-menu/render.h	2008-04-21 09:56:28.000000000 -0600
@@ -33,4 +33,6 @@ void hide_search(void);
 void measure_width(Menu_list_item * menu_item,int * max_width);
 gboolean _hide_all_windows(gpointer null);
 
+GtkWidget * build_menu_widget(Menu_item_color * mic, char * text,GdkPixbuf *pbuf,GdkPixbuf *pover,int max_width);
+
 #endif
diff -urpN awn-extras-applets-0.2.6.orig/src/cairo-menu-classic/backend-gnome.c awn-extras-applets-0.2.6/src/cairo-menu-classic/backend-gnome.c
--- awn-extras-applets-0.2.6.orig/src/cairo-menu-classic/backend-gnome.c	2008-02-14 09:49:53.000000000 -0700
+++ awn-extras-applets-0.2.6/src/cairo-menu-classic/backend-gnome.c	2008-04-20 23:22:01.000000000 -0600
@@ -34,6 +34,7 @@
 #include <libgen.h>
 #include <ctype.h>
 #include <libnotify/notify.h>
+#include <glib/gstdio.h>
 
 #include "menu_list_item.h"
 
--- awn-extras-applets-0.2.6.orig/src/cairo-menu/menu.h	2008-02-14 09:49:53.000000000 -0700
+++ awn-extras-applets-0.2.6/src/cairo-menu/menu.h	2008-04-21 10:13:22.000000000 -0600
@@ -58,6 +58,7 @@
 void fixed_move(GtkWidget *widget,gint x,gint y);
 void fixed_put(GtkWidget *widget,gint x,gint y);
 void hide_all_menus(void);
+GtkWidget * menu_new(GtkWidget * parent_menu);
 
 #endif 
 
