diff -Nur -x '*.orig' -x '*~' awn-extras-applets-0.2.6/src/places/applet.c awn-extras-applets-0.2.6.new/src/places/applet.c
--- awn-extras-applets-0.2.6/src/places/applet.c	2008-02-19 01:32:16.000000000 +0100
+++ awn-extras-applets-0.2.6.new/src/places/applet.c	2008-10-14 22:52:38.000000000 +0200
@@ -187,8 +187,16 @@
 static void config_get_color (AwnConfigClient *client, const gchar *key, AwnColor *color)
 {
 	gchar *value = awn_config_client_get_string (client, AWN_CONFIG_CLIENT_DEFAULT_GROUP, key, NULL);
-	awn_cairo_string_to_color (value, color);
-	g_free (value);
+	if (value)
+	{
+		awn_cairo_string_to_color (value, color);
+		g_free (value);
+	}
+	else
+	{
+		g_warning("Failed to read config key: %s\n",key);
+		awn_cairo_string_to_color ("000000", color); 
+	}
 }
 #else
 static void config_get_string (GConfClient *client, const gchar *key, gchar **str)
@@ -200,8 +208,16 @@
 static void config_get_color (GConfClient *client, const gchar *key, AwnColor *color)
 {
 	GConfValue *value = gconf_client_get_string (client, key, NULL);
-	awn_cairo_string_to_color (value, color);
-	g_free (value);
+	if (value)
+	{
+		awn_cairo_string_to_color (value, color);
+		g_free (value);
+	}
+	else
+	{
+		g_warning("Failed to read config key: %s\n",key);
+		awn_cairo_string_to_color ("000000", color); 
+	}
 }
 #endif
 
