diff -Nur -x '*.orig' -x '*~' awn-extras-applets-0.2.6/Makefile.schemas awn-extras-applets-0.2.6.new/Makefile.schemas
--- awn-extras-applets-0.2.6/Makefile.schemas	2008-02-14 17:49:53.000000000 +0100
+++ awn-extras-applets-0.2.6.new/Makefile.schemas	2008-10-15 00:10:28.000000000 +0200
@@ -10,7 +10,7 @@
 
 if DESKTOP_AGNOSTIC
 $(schemas_FILE).in: $(APPLET_SCHEMA)
-	$(AWN_BINDIR)/awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
+	awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
 endif
 
 @INTLTOOL_SCHEMAS_RULE@
diff -Nur -x '*.orig' -x '*~' awn-extras-applets-0.2.6/src/filebrowser/Makefile.in awn-extras-applets-0.2.6.new/src/filebrowser/Makefile.in
--- awn-extras-applets-0.2.6/src/filebrowser/Makefile.in	2008-10-15 00:03:45.000000000 +0200
+++ awn-extras-applets-0.2.6.new/src/filebrowser/Makefile.in	2008-10-15 00:06:33.000000000 +0200
@@ -720,7 +720,7 @@
 @INTLTOOL_DESKTOP_RULE@
 
 @DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@$(schemas_FILE).in: $(APPLET_SCHEMA)
-@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	$(AWN_BINDIR)/awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
+@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
 
 @GCONF_SCHEMAS_TRUE@@INTLTOOL_SCHEMAS_RULE@
 
diff -Nur -x '*.orig' -x '*~' awn-extras-applets-0.2.6/src/places/Makefile.in awn-extras-applets-0.2.6.new/src/places/Makefile.in
--- awn-extras-applets-0.2.6/src/places/Makefile.in	2008-10-15 00:03:45.000000000 +0200
+++ awn-extras-applets-0.2.6.new/src/places/Makefile.in	2008-10-15 00:07:29.000000000 +0200
@@ -698,7 +698,7 @@
 @INTLTOOL_DESKTOP_RULE@
 
 @DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@$(schemas_FILE).in: $(APPLET_SCHEMA)
-@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	$(AWN_BINDIR)/awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
+@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
 
 @GCONF_SCHEMAS_TRUE@@INTLTOOL_SCHEMAS_RULE@
 
diff -Nur -x '*.orig' -x '*~' awn-extras-applets-0.2.6/src/shiny-switcher/Makefile.in awn-extras-applets-0.2.6.new/src/shiny-switcher/Makefile.in
--- awn-extras-applets-0.2.6/src/shiny-switcher/Makefile.in	2008-10-15 00:03:45.000000000 +0200
+++ awn-extras-applets-0.2.6.new/src/shiny-switcher/Makefile.in	2008-10-15 00:08:04.000000000 +0200
@@ -705,7 +705,7 @@
 @INTLTOOL_DESKTOP_RULE@
 
 @DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@$(schemas_FILE).in: $(APPLET_SCHEMA)
-@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	$(AWN_BINDIR)/awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
+@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
 
 @GCONF_SCHEMAS_TRUE@@INTLTOOL_SCHEMAS_RULE@
 
diff -Nur -x '*.orig' -x '*~' awn-extras-applets-0.2.6/src/switcher/Makefile.in awn-extras-applets-0.2.6.new/src/switcher/Makefile.in
--- awn-extras-applets-0.2.6/src/switcher/Makefile.in	2008-10-15 00:03:45.000000000 +0200
+++ awn-extras-applets-0.2.6.new/src/switcher/Makefile.in	2008-10-15 00:09:19.000000000 +0200
@@ -699,7 +699,7 @@
 @INTLTOOL_DESKTOP_RULE@
 
 @DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@$(schemas_FILE).in: $(APPLET_SCHEMA)
-@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	$(AWN_BINDIR)/awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
+@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
 
 @GCONF_SCHEMAS_TRUE@@INTLTOOL_SCHEMAS_RULE@
 
diff -Nur -x '*.orig' -x '*~' awn-extras-applets-0.2.6/src/trash/Makefile.in awn-extras-applets-0.2.6.new/src/trash/Makefile.in
--- awn-extras-applets-0.2.6/src/trash/Makefile.in	2008-10-15 00:03:45.000000000 +0200
+++ awn-extras-applets-0.2.6.new/src/trash/Makefile.in	2008-10-15 00:08:47.000000000 +0200
@@ -737,7 +737,7 @@
 @INTLTOOL_DESKTOP_RULE@
 
 @DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@$(schemas_FILE).in: $(APPLET_SCHEMA)
-@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	$(AWN_BINDIR)/awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
+@DESKTOP_AGNOSTIC_TRUE@@GCONF_SCHEMAS_TRUE@	awn-schema-to-gconf $< $@ "$(APPLET_NAME)"
 
 @GCONF_SCHEMAS_TRUE@@INTLTOOL_SCHEMAS_RULE@
 
