diff -Nur -x '*.orig' -x '*~' awn-extras-applets-0.2.6/src/tsclient-applet/tsclient_app.py awn-extras-applets-0.2.6.new/src/tsclient-applet/tsclient_app.py
--- awn-extras-applets-0.2.6/src/tsclient-applet/tsclient_app.py	2008-02-19 01:32:16.000000000 +0100
+++ awn-extras-applets-0.2.6.new/src/tsclient-applet/tsclient_app.py	2008-10-15 03:00:45.000000000 +0200
@@ -29,7 +29,11 @@
     self.dialog.connect ("focus-out-event", self.dialog_focus_out)
     #a=dircache.listdir('/home/david/.tsclient')
     home = os.path.expanduser('~')
-    a=os.listdir(home + '/.tsclient')
+    try:
+    	a=os.listdir(home + '/.tsclient')
+    except OSError:
+    	os.mkdir(home + '/.tsclient')
+    	a=os.listdir(home + '/.tsclient')
     self.rootPath = home + "/.tsclient/"
     for item in a:
         (shortname, extension) = os.path.splitext(item)
