=== modified file 'src/volume-control/volume-control.py'
--- src/volume-control/volume-control.py	2008-09-15 20:49:22 +0000
+++ src/volume-control/volume-control.py	2008-09-15 20:53:58 +0000
@@ -209,6 +209,10 @@
         return self.channel
 
     def getVolume(self):
+        try:
+          return alsaaudio.Mixer(self.channel).getvolume()[0]
+        except:
+          self.setChannel()
         return alsaaudio.Mixer(self.channel).getvolume()[0]
 
     def setVolume(self, value):

