diff -ur awn-extras-applets-0.2.6.orig/src/comic/comicdialog.py awn-extras-applets-0.2.6/src/comic/comicdialog.py
--- awn-extras-applets-0.2.6.orig/src/comic/comicdialog.py	2008-03-15 17:12:04.000000000 +0100
+++ awn-extras-applets-0.2.6/src/comic/comicdialog.py	2008-03-15 18:35:47.000000000 +0100
@@ -31,7 +30,7 @@
 		self.connect("expose_event", self.expose)
 
 	def expose(self,widget,event):
-		icon_name = './dilbert.gif'
+		icon_name = '/tmp/dilbert.gif'
 		icon = gdk.pixbuf_new_from_file(icon_name)
 		dim = [icon.get_width(),icon.get_height()]
 		self.set_size_request(dim[0], dim[1])
Seulement dans awn-extras-applets-0.2.6/src/comic: comicdialog.py~
